vaadin-time-picker-flow-parent/vaadin-time-picker-flow/src/main/java/com/vaadin/flow/component/timepicker/GeneratedVaadinTimePicker.java

```diff
@@ -937,10 +937,12 @@ public GeneratedVaadinTimePicker(T initialValue, T defaultValue,
     public <P> GeneratedVaadinTimePicker(T initialValue, T defaultValue,
             Class<P> elementPropertyType,
             SerializableBiFunction<R, P, T> presentationToModel,
-            SerializableBiFunction<R, T, P> modelToPresentation) {
+            SerializableBiFunction<R, T, P> modelToPresentation,
+            boolean isInitialValueOptional) {
         super("value", defaultValue, elementPropertyType, presentationToModel,
                 modelToPresentation);
-        if (initialValue != null) {
+        if ((getElement().getProperty("value") == null
+                || !isInitialValueOptional) && initialValue != null) {
```

Comment:
Needed to change this constructor to support the `isInitialValueOptional` option. The implementation is copied from the other constructor that was previously used by `TimePicker`.